Source/WebCore/Modules/model-element/HTMLModelElement.cpp

```diff
@@ -688,6 +728,73 @@ void HTMLModelElement::setCurrentTime(double currentTime)
     if (m_modelPlayer)
         m_modelPlayer->setCurrentTime(Seconds(currentTime), [&]() { });
 }
+
+const URL& HTMLModelElement::environmentMap() const
+{
+    return m_environmentMapURL;
+};
+
+void HTMLModelElement::setEnvironmentMap(const URL& url)
+{
+    if (url.string() == m_environmentMapURL.string())
+        return;
+
+    m_environmentMapURL = url;
+
+    m_environmentMapData.reset();
+
+    if (m_environmentMapResource) {
+        m_environmentMapResource->removeClient(*this);
+        m_environmentMapResource = nullptr;
+    }
+
+    if (!m_environmentMapReadyPromise->isFulfilled())
+        m_environmentMapReadyPromise->reject(Exception { ExceptionCode::AbortError });
+
+    m_environmentMapReadyPromise = makeUniqueRef<EnvironmentMapPromise>();
+
+    if (m_environmentMapURL.isEmpty()) {
+        ActiveDOMObject::queueTaskToDispatchEvent(*this, TaskSource::DOMManipulation, Event::create(eventNames().errorEvent, Event::CanBubble::No, Event::IsCancelable::No));
```

Comment:
I don't think we should fire the error event here - wouldn't it conflict with the error event from regular model resource loading?

Reply:
Removed events sending, let's leave it to the spec to decide when and what events we send. Also, empty URL is a valid value that indicates removal of IBL, added corresponding code for that.